pkg/auth/manager_test.go

```diff
@@ -138,14 +149,33 @@ func (r *fakeAuthHandler) subscribeToRotatedIdentities() <-chan certs.Certificat
 
 // Fake AuthMap
 type fakeAuthMap struct {
-	entries map[authKey]authInfo
+	entries    map[authKey]authInfo
+	failDelete bool
 }
 
 func (r *fakeAuthMap) Delete(key authKey) error {
+	if r.failDelete {
+		return errors.New("failed to delete entry")
+	}
+
 	delete(r.entries, key)
 	return nil
 }
 
+func (r *fakeAuthMap) DeleteIf(predicate func(key authKey, info authInfo) bool) error {
+	if r.failDelete {
+		return errors.New("failed to delete entry")
+	}
+
+	for k, v := range r.entries {
```

Comment:
Could codegolf this to: `maps.DeleteFunc(r.entries, predicate)` ;-)

Reply:
@joamaki changed, thanks